tensorboard/webapp/plugins/plugins.component.ts

```diff
@@ -0,0 +1,117 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+
+import {UiPluginMetadata} from './plugins.container';
+import {
+  LoadingMechanismType,
+  CustomElementLoadingMechanism,
+} from '../types/api';
+
+@Component({
+  selector: 'plugins-component',
+  templateUrl: './plugins.component.html',
+  styles: [
+    '.plugins { height: 100%; }',
+    'iframe { border: 0; height: 100%; width: 100%; }',
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PluginsComponent implements OnChanges {
+  @ViewChild('pluginContainer', {static: true, read: ElementRef})
+  private readonly pluginsContainer!: ElementRef<HTMLDivElement>;
+
+  @Input()
+  activePlugin?: UiPluginMetadata;
+
+  @Input()
+  lastUpdated?: number;
+
+  readonly LoadingMechanismType = LoadingMechanismType;
+
+  private readonly pluginInstances = new Map<string, HTMLElement>();
+
+  ngOnChanges(change: SimpleChanges): void {
+    if (change.activePlugin && this.activePlugin) {
+      this.renderPlugin(this.activePlugin!);
+    }
+    if (change.lastUpdated) {
+      this.reload();
+    }
+  }
+
+  private renderPlugin(plugin: UiPluginMetadata) {
+    for (const element of this.pluginInstances.values()) {
+      element.style.display = 'none';
+    }
+
+    if (this.pluginInstances.has(plugin.id)) {
+      const instance = this.pluginInstances.get(plugin.id) as HTMLElement;
+      instance.style.display = null;
+      return;
+    }
+
+    this.appendPlugin(plugin);
+  }
+
+  private appendPlugin(plugin: UiPluginMetadata) {
+    const pluginId = plugin.id;
```

Comment:
How about do a check here: if `this.pluginInstance` already contains `pluginId`, throw an Error?

Reply:
Took a stab at addressing your feedback differently. PTAL :)